src/logic/calculate.js

```diff
@@ -0,0 +1,34 @@
+import operate from './operate';
+
+const calculate = (calculatorData, buttonName) => {
+  const dataUpdate = { ...calculatorData };
```

Comment:
- [ ] `dataUpdate` object is the same `calculatorData` object, you are assigning a new variable to the same variable without any modification. Alternatively we can de-structure `calculatorData` object, to avoid repititive call to `calculatorData`:

```js
const { total, next, operation } = calculatorData;
```